### PR TITLE
add docker observer component

### PR DESCRIPTION
**Pull request by mstumpfx**
Adds the docker observer extension to the distribution

### CODE REVIEW

**Comment by rmfitzpatrick** [comment]
Should this be gated by https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6665?

File: go.mod [hunk 1]
```diff
@@ -80,6 +80,8 @@ require (
 	gopkg.in/yaml.v2 v2.4.0
 )
 
+require github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer/dockerobserver v0.41.0
```
Comment (rmfitzpatrick) [inline go.mod:83]: nit, mind adding to require block above?

Comment (mstumpfx) [reply go.mod]: Done, thanks. I assumed `go mod tidy` would have put this in place for me.

**Comment by mstumpfx** [comment]
> Should this be gated by [open-telemetry/opentelemetry-collector-contrib#6665](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6665)?

_Probably_ - it will be effectively unusable until then.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### 🚀 New components 🚀
+
+- [`docker observer`](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/observer/dockerobserver) to detect and create container endpoints, to be used with the [receiver creator](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/receivercreator).
```
Comment (rmfitzpatrick) [inline CHANGELOG.md:7]:
```suggestion
- [`docker_observer`](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/observer/dockerobserver) to detect and create container endpoints, to be used with the [`receiver_creator`](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/receivercreator).
```
?

**Review by rmfitzpatrick — APPROVED** [review]

**Pull request merged**